src/libsyntax/tokenstream.rs

```diff
@@ -196,6 +201,10 @@ impl TokenStream {
         }
     }
 
+    pub fn builder() -> TokenStreamBuilder {
```

Comment:
I think this would be better as a free function rather than a method on TokenStream